File: iteration_4-16/Iteration.java
import java.util.Scanner;

public class Iteration {
    public static void main(String[] args) {
        breakDownInput(getPositiveInt());

    }

    public static int getPositiveInt() {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter a positive, whole number:");
        while (!input.hasNextInt()) {
            System.out.println("Please enter a positive, whole number.");
            input.nextLine();
        }
        int inputNumber = input.nextInt();
        input.close();
        return inputNumber;
    }

    public static void breakDownInput(int input) {
        int newInput = input - 1;
        while (newInput > 0) {
            System.out.println(input - newInput + " " + isFactorOf(input, (input - newInput)));
            newInput -= 1;
        }
        System.out.println("Done");
        // for (int i = 1; i < input; i++) {
        // System.out.println(i + " " + isFactorOf(input, i));
        // }
        // System.out.println("Done");
    }

    public static String isFactorOf(int input, int newInput) {
        if (input % newInput == 0) {
            return "Is a factor of " + input;
        } else {
            return "Is not a factor of " + input;
        }
    }
}

File: iteration_4-16/iteration.py
def get_input():
    user_input = input("Please enter a number:")
    return int(user_input)


def break_down_input(user_input):
    # new_input = user_input - 1
    # while new_input > 0:
    #     print(
    #         str(user_input - new_input) + " " +
    #         is_factor_of(user_input, (user_input - new_input)))
    #     new_input -= 1
    i = 1
    while (i < user_input):
        print(str(i) + " " + is_factor_of(user_input, i))
        i += 1
    print("Done")


def is_factor_of(user_input, new_input):
    if user_input % new_input == 0:
        return "is a factor of " + str(user_input)
    else:
        return "is not a factor of " + str(user_input)


def main():
    break_down_input(get_input())


if __name__ == "__main__":
    main()
